scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -419,7 +419,8 @@ abstract class FixedPathSource(path: String*) extends FileSource {
   def localPaths = path.toList
   def hdfsPaths = path.toList
 
-  override def toString = getClass.getName + path
+  // The collection type of `path` causes equality issues in JobTest
```

Comment:
I hadn't noticed what was doing this before, its really that the equals relies on toString (usually toString really is just for human's and doesn't matter.). The change looks good to me, but you change this to mention that equals is relying on toString and path.toString isn't stable ordinarily without this.

I suspect the underlying problem here is that the type of the var arg's string is probably something like a Seq[String], and the 2 means of creation in your test example here provide underlying different classes. toString uses toString on the elements which winds up calling getClass, so is printing out the actual implementations passed.